x-pack/plugins/security_solution/common/api/timeline/model/components.gen.ts

```diff
@@ -0,0 +1,350 @@
+  favorite: z.array(FavoriteTimelineResult).optional(),
+  filters: z.array(FilterTimelineResult).optional(),
+  kqlMode: z.string().optional(),
+  kqlQuery: SerializedFilterQueryResult.optional(),
+  indexNames: z.array(z.string()).optional(),
+  savedQueryId: z.string().optional(),
+  sort: Sort.optional(),
+  status: z.enum(['active', 'draft', 'immutable']).optional(),
+  title: z.string().optional(),
+  templateTimelineId: z.string().optional(),
+  templateTimelineVersion: z.number().optional(),
+  timelineType: TimelineType.optional(),
+  updated: z.number().optional(),
+  updatedBy: z.string().optional(),
+});
+
+export type BareNote = z.infer<typeof BareNote>;
+export const BareNote = z.object({
+  eventId: z.string().optional(),
+  note: z.string().optional(),
+  timelineId: z.string().optional(),
+  created: z.number().optional(),
+  createdBy: z.string().optional(),
+  updated: z.number().optional(),
+  updatedBy: z.string().optional(),
+});
+
+export type Note = z.infer<typeof Note>;
+export const Note = BareNote.merge(
+  z.object({
+    noteId: z.string().optional(),
+    version: z.string().optional(),
+  })
+);
+
+export type PinnedEvent = z.infer<typeof PinnedEvent>;
+export const PinnedEvent = z.object({
+  pinnedEventId: z.string().optional(),
+  eventId: z.string().optional(),
+  timelineId: z.string().optional(),
+  created: z.number().optional(),
+  createdBy: z.string().optional(),
+  updated: z.number().optional(),
+  updatedBy: z.string().optional(),
+  version: z.string().optional(),
+});
+
+export type TimelineResponse = z.infer<typeof TimelineResponse>;
+export const TimelineResponse = SavedTimeline.merge(
+  z.object({
+    eventIdToNoteIds: z.array(Note).optional(),
+    notes: z.array(Note).optional(),
+    noteIds: z.array(z.string()).optional(),
+    pinnedEventIds: z.array(z.string()).optional(),
+    pinnedEventsSaveObject: z.array(PinnedEvent).optional(),
+    savedObjectId: z.string(),
+    version: z.string(),
+  })
+);
+
+export type FavoriteTimelineResponse = z.infer<typeof FavoriteTimelineResponse>;
+export const FavoriteTimelineResponse = z.object({
+  savedObjectId: z.string(),
+  version: z.string(),
+  code: z.number().nullable().optional(),
+  message: z.string().nullable().optional(),
+  templateTimelineId: z.string().nullable().optional(),
+  templateTimelineVersion: z.number().nullable().optional(),
+  timelineType: TimelineType.optional(),
+  favorite: z.array(FavoriteTimelineResult).optional(),
+});
+
+export type GlobalNote = z.infer<typeof GlobalNote>;
+export const GlobalNote = z.object({
+  noteId: z.string().optional(),
+  version: z.string().optional(),
+  note: z.string().optional(),
+  timelineId: z.string().optional(),
+  created: z.number().optional(),
+  createdBy: z.string().optional(),
+  updated: z.number().optional(),
+  updatedBy: z.string().optional(),
+});
+
+/**
+ * The field to sort the timelines by.
+ */
+export type SortFieldTimeline = z.infer<typeof SortFieldTimeline>;
```

Comment:
This should be an enum as seen [here](https://github.com/elastic/kibana/blob/c325e3caf2eca378dee077dd421bd0defeb1a48a/x-pack/plugins/security_solution/common/api/timeline/model/api.ts#L493) ..the changes in my PR [here](https://github.com/elastic/kibana/pull/187172/files#diff-4d228c7af32f2c32b998410c27daa4dd90784145c284ef3eded415f2256b06d2R242)